### PR TITLE
Tron support

### PR DESCRIPTION
**Pull request by tmcgroul**

### CODE REVIEW

File: sqa/tron/query.py [hunk 1]
```diff
@@ -0,0 +1,451 @@
...
+
+class BlockFieldSelection(TypedDict, total=False):
+    number: bool
+    hash: bool
+    parentHash: bool
+    timestamp: bool
+    txTrieRoot: bool
+    version: bool
+    timestamp: bool
+    witness_address: bool
```
Comment (eldargab) [inline sqa/tron/query.py:19]: Why not camel case?

File: sqa/tron/query.py [hunk 2]
```diff
@@ -0,0 +1,451 @@
...
+    'energyUsageTotal': bool,
+    'netUsage': bool,
+    'netFee': bool,
+    'originEnergyUsage': bool,
+    'energyPenaltyTotal': bool,
+}, total=False)
+
+
+class LogFieldSelection(TypedDict, total=False):
+    logIndex: bool
```
Comment (eldargab) [inline sqa/tron/query.py:56]: Let's remove it to match rest networks. `transaction_index` should be added as a non-key field.

**Review by eldargab — CHANGES_REQUESTED** [review]

File: sqa/tron/writer/parquet.py [hunk 3]
```diff
@@ -0,0 +1,309 @@
...
+        logs,
+        row_group_size=10_000,
+        use_dictionary=['address', 'topic0'],
+        write_statistics=['_idx', 'address', 'topic0', 'block_number'],
+        **kwargs
+    )
+
+    transactions = tables['transactions']
+    transactions = transactions.sort_by([
+        ('type', 'ascending'),
```
[the remark this answers is not in the document]
Comment (tmcgroul) [reply sqa/tron/writer/parquet.py:274]: not every type of transaction has root contract. still it's better to sort by `tx.contract_address`?

File: sqa/tron/query.py [hunk 4]
```diff
@@ -0,0 +1,451 @@
...
+    primary_key=['log_index'],
+    column_weights={
+        'data': 'data_size'
+    }
+)
+
+
+_internal_tx_table = Table(
+    name='internal_transactions',
+    primary_key=['hash']
```
[the remark this answers is not in the document]
Comment (tmcgroul) [reply sqa/tron/query.py:214]: should `pos` represent position relatively to its transaction or block?
in evm `logIndex` points to log position amount others logs in a block and here logs follow the same behaviour

File: sqa/tron/writer/parquet.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (eldargab) [reply sqa/tron/writer/parquet.py:272]: The point is to sort the data in such a way, that rows matching typical filters would tend to be together.

You can sort by multiple fields, e.g.

```
type
address_when_type_is_1
address_when_type_is_2
etc
```

File: sqa/tron/query.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (eldargab) [reply sqa/tron/query.py]: Relative to transaction counting is definitely more useful...

File: sqa/tron/query.py [hunk 5]
```diff
@@ -0,0 +1,444 @@
...
+    primary_key=['transaction_index'],
+    column_weights={
+        'raw_data_hex': 'raw_data_hex_size'
+    }
+)
+
+
+_logs_table = Table(
+    name='logs',
+    primary_key=['transaction_index', 'log_index'],
```
Comment (eldargab) [inline sqa/tron/query.py:200]: For logs table it is better to be compatible with EVM?

Comment (tmcgroul) [reply sqa/tron/query.py:200]: i believe it is up to us. wouldn't it be confusing to have one entity (internal tx) indexed relatively to tx and another one (log) relatively to block?

**Pull request merged**